centos_ci.sh

```diff
@@ -208,27 +205,101 @@ function docs_tar_upload() {
   echo "Find docs tar here http://artifacts.ci.centos.org/minishift/minishift/docs/$LATEST."
 }
 
-if [[ "$UID" = 0 ]]; then
+function create_release_commit() {
+  # Set Terminal
+  export TERM=xterm-256color
+  # Add git a/c identity
+  git config user.email "minishift@lists.minishift.io"
+  git config user.name "Minishift Bot"
+  # Export GITHUB_ACCESS_TOKEN
+  export GITHUB_ACCESS_TOKEN=$GITHUB_TOKEN
+  # Create master branch as git clone in CI doesn't create it
+  git checkout -b master
+  # Bump version and commit
+  sed -i "s|MINISHIFT_VERSION = .*|MINISHIFT_VERSION = $RELEASE_VERSION|" Makefile
+  git add Makefile
+  git commit -m "cut v$RELEASE_VERSION"
+  # Push update
+  # TODO: Change to minishift once final review is done
+  git push https://$1:$GITHUB_TOKEN@github.com/$1/minishift master
+}
+
+function add_release_notes() {
+  # TODO: Change to minishift once final review is done
+  release_id=$(curl -s "https://api.github.com/repos/${1}/minishift/releases" | jq --arg release "v$RELEASE_VERSION" -r ' .[] | if .name == $release then .id else empty end')
+
+  if [[ "$release_id" != "" ]]; then
+    # Generate required json payload for release note
+    ./scripts/release/issue-list.sh -r minishift -m $MILESTONE_ID | jq -Rn 'inputs + "\n"' | jq -s '{body:  add }' > json_payload.json
+    # Add release notes
+    # TODO: Change to minishift once final review is done
+    curl -H "Content-Type: application/json" -H "Authorization: token $GITHUB_TOKEN" \
+         --data @json_payload.json https://api.github.com/repos/${1}/minishift/releases/$release_id
+
+    # TODO: Change to minishift once final review is done
+    echo "Release notes of Minishift v$RELEASE_VERSION has been successfully updated. Find the release notes here https://github.com/${1}/minishift/releases/tag/v$RELEASE_VERSION."
+  else
+    echo "Failed to update release notes of Minishift v$RELEASE_VERSION as couldn't find the release ID. Try to manually update the release notes here https://github.com/${1}/minishift/releases/tag/v$RELEASE_VERSION."
+    exit 1
+  fi
+}
+
+
+function setup_build_environment() {
   load_jenkins_vars;
   prepare_ci_user;
   install_core_deps;
   install_kvm_virt;
   install_docker;
   runuser -l minishift_ci -c "/bin/bash centos_ci.sh"
+}
+
+function perform_release() {
+  GIT_USER="minishift-bot"
+  cd gopath/src/github.com/minishift/minishift
+  create_release_commit $GIT_USER;
+
+  make release
+
+  if [[ "$?" = 0 ]]; then
+    echo "Minishift v$RELEASE_VERSION has been successfully released. Find the latest release here https://github.com/$GIT_USER/minishift/releases/tag/v$RELEASE_VERSION."
+  else
+    echo "Failed to release Minishift v$RELEASE_VERSION. Try to release manually."
+    exit 1
+  fi
+
+  add_release_notes $GIT_USER;
```

Comment:
I guess I am still not clear how Jenkins handles non zero return values in these scripts. Does it fail the build on the first non zero return or is only the return value of the last command in the script relevant. I would think the latter, right?